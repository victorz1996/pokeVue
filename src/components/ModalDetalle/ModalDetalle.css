.botonModal {
    background: #f22539;
    border-radius: 10rem;
    color: aliceblue;
    height: 40px;
    width: 170px;
    margin-right: 10px;
    margin-left: 10px;
}
.iconoModal{
    position: absolute;
    right: 20px;
}
.btn-circleModal {
    width: 40px;
    height: 40px;
    padding: 6px 0px;
    border-radius: 20px;
    text-align: center;
    font-size: 12px;
    background: rgb(238, 236, 234);
    line-height: 1.42857;
  }