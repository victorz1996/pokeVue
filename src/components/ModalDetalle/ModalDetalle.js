/**
 * @namespace ModalDetalle
 * @description Componente que contiene la informacion que se muestra en el modal
 * de infromacion detallada de cada pokemon.
 * @author       Victor Zabala <victorma164@gmail.com>
 */
export default {
    name: "ModalDetalle",
    props: {
        infoPoke: { type: Object },
        pokesFavoritos: { type: Array }
    },
    data() {
        return {
            msn: '',
            esFavorito: false,
            portapapel: '',
        }
    },
    watch: {
        /**
        * @memberOf ModalDetalle
        * @function infoPoke Watch de infoPoke
        * @description Metodo que se ejecuta cada vez que la variable de infoPoke cambia
        */
        infoPoke() {
            this.obtenerTipos()
            const arrayMsn = this.msn.split(',')
            arrayMsn.splice(arrayMsn.length - 1, 1)
            this.msn = arrayMsn.toString()
            this.encontrarFavorito()
            this.modificarPortapapel()
        }
    },
    methods: {
        /**
        * @memberOf ModalDetalle
        * @function onCopy copiar portapapeles
        * @description Metodo que se ejecuta para copiar informacion de un pokemon al
        * portapapeles.
        * @param {Object} e
        */
        onCopy(e) {
            alert(`Copiado al portapapeles exitosamente: \n${e.text}`)
        },
        /**
        * @memberOf ModalDetalle
        * @function onError Error portapapeles
        * @description Metodo que se ejecuta cuando ocurre un error en la copia de informacion
        * al portapapeles
        */
        onError() {
            alert('FallÃ³ la copia')
        },
        /**
        * @memberOf ModalDetalle
        * @function modificarPortapapel modifica portapapel
        * @description Metodo que se ejecuta para crear el string con la informacion que se copiara
        * al portapapeles
        */
        modificarPortapapel() {
            if (this.infoPoke) {
                this.portapapel = `Name: ${this.infoPoke.forms[0].name}\n Weight: ${this.infoPoke.weight}\n Height: ${this.infoPoke.height}\n Types: ${this.msn}`
            }
        },
        /**
        * @memberOf ModalDetalle
        * @function encontrarFavorito encontrar pokemon favorito
        * @description Metodo que se ejecuta para identificar en la informacion detallada
        * si el pokemon es favorito
        */
        encontrarFavorito() {
            if (this.pokesFavoritos && this.pokesFavoritos.length) {
                const existe = this.pokesFavoritos.find(poke => poke.name === this.infoPoke.forms[0].name)
                if (!existe) {
                    this.esFavorito = false
                } else {
                    this.esFavorito = true
                }
            }
        },
        /**
        * @memberOf ModalDetalle
        * @function obtenerTipos encontrar pokemon tipos
        * @description Metodo que se ejecuta para organizar la informacion que se 
        * mostrara en los tipos del pokemon
        */
        obtenerTipos() {
            this.msn = ''
            if (this.infoPoke) {
                this.infoPoke.types.forEach(item => {
                    this.msn += `${item.type.name},`
                });
            }
        }
    },
}