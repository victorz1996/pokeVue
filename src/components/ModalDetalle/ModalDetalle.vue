<template>
  <div v-if="infoPoke">
    <ul class="list-group list-group-flush mb-4">
      <li style="text-transform: capitalize" class="list-group-item">
        <strong>Name: </strong> {{ infoPoke.forms[0].name }}
      </li>
      <li class="list-group-item">
        <strong>Weght: </strong> {{ infoPoke.weight }}
      </li>
      <li class="list-group-item">
        <strong>Height: </strong> {{ infoPoke.height }}
      </li>
      <li style="text-transform: capitalize" class="list-group-item">
        <strong>Types: </strong> {{ msn }}
      </li>
    </ul>
    <button
      v-clipboard:success="onCopy"
      v-clipboard:error="onError"
      v-clipboard:copy="portapapel"
      class="btn botonModal"
    >
      Share to my friends
    </button>
    <button class="btn btn-circleModal iconoModal">
      <font-awesome-icon
        v-show="esFavorito"
        style="color: rgb(228, 228, 9)"
        icon="star"
      />
      <font-awesome-icon
        v-show="!esFavorito"
        style="color: rgb(104, 103, 103)"
        icon="star"
      />
    </button>
  </div>
</template>

<script src="./ModalDetalle.js"></script>
<style lang="css" src="./ModalDetalle.css"></style>