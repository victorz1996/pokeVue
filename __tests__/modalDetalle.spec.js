import { shallowMount } from '@vue/test-utils'
import ModalDetalle from "../src/components/ModalDetalle/ModalDetalle.vue"

let cmp

    // eslint-disable-next-line no-undef
    beforeEach(() => {
      cmp = shallowMount(ModalDetalle)
    })
    // eslint-disable-next-line no-undef
    describe('Componente ModalDetalle', () => {
      // eslint-disable-next-line no-undef
      test('Instancia de Vue.js',() => {
        // eslint-disable-next-line no-undef
        expect(cmp.isVueInstance).toBeTruthy()
      })

      // eslint-disable-next-line no-undef
      it('HTMl renderizado structure esperada',()=>{
        // eslint-disable-next-line no-undef
        expect(cmp.element).toMatchSnapshot()
    })
})