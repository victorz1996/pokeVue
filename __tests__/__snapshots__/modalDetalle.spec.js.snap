// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente ModalDetalle HTMl renderizado structure esperada 1`] = `<!---->`;
